### ISSUE TITLE
Using adjust_lr in combination with split_sizes will give the wrong learning rate.

### ISSUE THREAD

**Issue by skaae** [microsoft/dion#111]
I have noticed that if i enable split_sizes in NorMuon in order to split a gated MLP and also enable `adjust_lr="rmsnorm"` my training runs will initially converge faster but then diverge.

I think there might be a bug here: https://github.com/microsoft/dion/blob/0a059a4542e5aa74b40dcd1dce363836c7afcd48/dion/normuon.py#L329

where the the learning rate adjustment does not take into account the reduced sizes specified with split_sizes? This would result in too high learning rate for split matrices. 
I don't completely follow the `normuon_normalization_stacked` but i don't think the gradients are normalized here? 

BR Søren

**Comment by JohnLangford** [comment]
I found and addressed an issue related to this, but perhaps not the issue you have in mind.   To get at that, perhaps more info would be needed?  Do you have a shareable example?

**Comment by skaae** [comment]
Hi John, 
Thank you for the reply. My question is if the current code take `split_sizes` into account when calculating the learning rate adjustment or muon? I'm asking because I don't see split size go into `adjusted_lr = adjust_lr_rms_norm(lr, X[0].shape, flatten=flatten) `, but maybe the gradient is scaled elsewhere to take `split_size` into account? 

Example. Say i have  gatedMLP implemented as   
```
self.fc = nn.Linear(in_dim, 2*hidden_dim)

def forward(x):
    h = self.fc(x)
    gate, linear = h.chunk(2, dim=-1)
    return F.silu(gate) * linear
```
Will the current implementation then correctly use `hidden_size` as the dim instead of `2*hidden_size` when calculating the rms norm lr adjustment?

**Comment by JohnLangford** [comment]
Yes after PR #113/#114.
